### PR TITLE
Support max versions in typeshed VERSIONS file

### PR DESCRIPTION
**Pull request by srittau**
As discussed in python/typeshed#5304. Please note that this PR is mostly untested, but I hope CI passes.

### CODE REVIEW

**Comment by rchen152** [comment]
Thanks! Looks pretty good to me, but let's see what the tests say. (Seems that workflows now need approval from a maintainer to run if a PR is from a first-time contributor...)

**Comment by srittau** [comment]
Ah, found the tests. 😃 Will fix and provide tests for the new functionality.

File: pytype/pytd/typeshed.py [hunk 1]
```diff
@@ -104,20 +104,26 @@ def _load_stdlib_versions(self):
...
 
     Returns:
-      A mapping from module name to (major, minor) Python version.
+      A mapping from module name to ((min_major, min_minor),
+      (max_major, max_minor)) Python version. The max tuple can be `None`.
     """
     _, text = self._load_file(os.path.join("stdlib", "VERSIONS"))
     versions = {}
     for line in text.splitlines():
-      match = re.fullmatch(r"(.+): (\d)\.(\d+)", line)
```
Comment (JelleZijlstra) [inline pytype/pytd/typeshed.py:117]: While you're at it could you add support for skipping blank and comment lines here, like mypy does?

**Comment by srittau** [comment]
Since the current tests also check against the real `VERSIONS` file, I think it's easier to wait for that file to be updated using the new syntax before adding tests.

**Review by JelleZijlstra — APPROVED** [review]

**Review by rchen152 — APPROVED** [review]
Thanks again! Importing this into our internal codebase now.

**Pull request closed without merging**